pkg/kv/kvserver/client_replica_circuit_breaker_test.go

```diff
@@ -886,6 +888,19 @@ func (cbt *circuitBreakerTest) UntripsSoon(t *testing.T, method func(idx int) er
 	})
 }
 
+func (cbt *circuitBreakerTest) WaitForProposals(t *testing.T, idx int) {
+	t.Helper()
+	testutils.SucceedsSoon(t, func() error {
+		t.Helper()
+
+		repl := cbt.repls[idx].Replica
+		if n := repl.NumPendingProposals(); n > 0 {
```

Comment:
I wonder if this can flap between 0 and non-0 for a bit. Depends on whether the sum of reproposals map size and propbuf size that `NumPendingProposals` returns is "exhaustive", i.e. there are no sneaky (re-)proposals in some transition state that is neither in the map nor in the buffer.

Don't need to spend a ton of time figuring this out now though.

Reply:
We're holding the replica mutex across these, so I think they should be in sync.

Reply:
I mean "transitions" in between holding mutexes. I.e. does the invariant "all possible (re-)proposals are either in proposals map or in proposals buffer" hold any time we grab replica mu? I vaguely remember there were cases when we delete and reinsert things in the map.

Reply:
>I vaguely remember there were cases when we delete and reinsert things in the map.

I believe those cases happen under lock.